apps-script/incident-response/ChatApp.gs

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// [START chat_incident_response_app]
+
+/**
+ * Responds to a MESSAGE event in Google Chat.
+ *
+ * This app only responds to a slash command with the ID 1 ("/closeIncident").
+ * It will respond to any other message with a simple "Hello" text message.
+ *
+ * @param {Object} event the event object from Google Chat
+ */
+function onMessage(event) {
+  if (event.message.slashCommand) {
+    return processSlashCommand_(event);
+  }
+  return { "text": "Hello from Incident Response app!" };
+}
+
+/**
+ * Responds to a CARD_CLICKED event in Google Chat.
+ *
+ * This app only responds to one kind of dialog (Close Incident).
+ *
+ * @param {Object} event the event object from Google Chat
+ */
+function onCardClick(event) {
+  if (event.isDialogEvent) {
+    if (event.dialogEventType == 'SUBMIT_DIALOG') {
+      return processSubmitDialog_(event);
+    }
+    return {
+      actionResponse: {
+        type: "DIALOG",
+        dialogAction: {
+          actionStatus: "OK"
+        }
+      }
+    };
+  }
+}
+
+/**
+ * Responds to a MESSAGE event with a Slash command in Google Chat.
+ *
+ * This app only responds to a slash command with the ID 1 ("/closeIncident")
+ * by returning a Dialog.
+ *
+ * @param {Object} event the event object from Google Chat
+ */
+function processSlashCommand_(event) {
+  if (event.message.slashCommand.commandId != CLOSE_INCIDENT_COMMAND_ID) {
+    return {
+      "text": "Command not recognized. Use the command `/closeIncident` to close the incident managed by this space."
+    };
+  }
+  const sections = [
+    {
+      header: "Close Incident",
+      widgets: [
+        {
+          textInput: {
+            label: "Please describe the incident resolution",
+            type: "MULTIPLE_LINE",
+            name: "description"
+          }
+        },
+        {
+          buttonList: {
+            buttons: [
+              {
+                text: "Close Incident",
+                onClick: {
+                  action: {
+                    function: "closeIncident"
+                  }
+                }
+              }
+            ]
+          }
+        }
+      ]
+    }
+  ];
+  return {
+    actionResponse: {
+      type: "DIALOG",
+      dialogAction: {
+        dialog: {
+          body: {
+            sections: sections
```

Comment:
nit: I typically prefer the [shorthand](https://google.github.io/styleguide/jsguide.html#features-objects-shorthand-properties)

```suggestion
            sections,
```